tools/clang/lib/SPIRV/DeclResultIdMapper.cpp

```diff
@@ -23,10 +23,34 @@
 #include "llvm/ADT/StringMap.h"
 #include "llvm/ADT/StringSet.h"
 
+#include "SPIRVEmitter.h"
+
 namespace clang {
 namespace spirv {
 
 namespace {
+/// \brief Returns true if the given decl is a boolean stage I/O variable.
+/// Returns false if the type is not boolean, or the decl is a built-in stage
+/// variable.
+bool isBooleanStageIOVar(const NamedDecl *decl, QualType type,
+                         const hlsl::DXIL::SemanticKind semanticKind,
+                         const hlsl::SigPoint::Kind sigPointKind) {
+  // [[vk::builtin(...)]] makes the decl a built-in stage variable.
```

Comment:
This is actually incorrect. `[[vk::builtin]]` is a general mechanism; it's just that the current supported cases are all booleans. We should at least put an `assert` here.

Reply:
Er, sorry, ignore this. I'm silly.